pages/interface/components/Markdown/plugins/anchor-headers.js

```diff
@@ -0,0 +1,104 @@
+import { getSlug } from 'utils/slug';
+
+let isListenerAdded = false;
+
+/**
+ * @returns {import('bytemd').BytemdPlugin}
+ */
+export function anchorHeadersPlugin() {
+  function appendAnchorToHeader(header) {
+    if (header.children.length === 1 && isAnchor(header.children[0])) {
+      let linkElement = header.children[0];
+      if (linkElement.innerText === header.innerText) {
+        return linkElement;
+      }
+    }
+
+    const anchor = document.createElement('a');
+    anchor.innerHTML = header.innerHTML;
+    header.innerText = '';
+    header.appendChild(anchor);
+    return anchor;
+  }
+
+  function isAnchor(element) {
+    return element.tagName.toLowerCase() === 'a';
+  }
+
+  function sanitizeId(id) {
+    const prefix = 'user-content-';
+    return id.startsWith(prefix) ? id : `${prefix}${id}`;
+  }
+
+  function desanitizeId(id) {
+    return id.replace(/^user-content-/, '');
+  }
+
+  function handleClick(event) {
+    let targetAnchor = event.target;
+    while (targetAnchor && !isAnchor(targetAnchor)) {
+      targetAnchor = targetAnchor.parentNode;
+    }
+
+    if (!targetAnchor) {
+      return;
+    }
+
+    const currentPage = window.location.href.split('#')[0];
+    const targetPage = targetAnchor.href.split('#')[0];
+    if (targetPage !== currentPage) {
+      return;
+    }
+
+    scrollToHash(targetAnchor.hash);
+  }
+
+  function scrollToHash(encodedHash) {
+    let hash;
+    try {
+      hash = decodeURIComponent(encodedHash.slice(1)).toLowerCase();
+    } catch {
+      return;
+    }
+
+    const target = document.getElementById(sanitizeId(hash));
+    if (target) {
+      target.scrollIntoView();
+    }
+    return !!target;
+  }
+
+  let shouldScroll = false;
+  if (!isListenerAdded && typeof window !== 'undefined') {
+    window.addEventListener('load', () => {
+      shouldScroll = !!location.hash;
+    });
+  }
+
+  return {
+    viewerEffect({ markdownBody }) {
+      const headers = markdownBody.querySelectorAll('h1, h2, h3, h4, h5, h6');
+
+      for (const header of headers) {
+        if (!header.innerText) continue;
+
+        if (!header.id) {
+          header.setAttribute('id', sanitizeId(getSlug(header.innerText)));
+        }
+
+        const anchor = appendAnchorToHeader(header);
+        anchor.href = `#${desanitizeId(header.id)}`;
+        anchor.setAttribute('tabindex', -1);
+        anchor.addEventListener('click', handleClick);
+
+        const iconDiv = document.createElement('div');
+        iconDiv.className = 'octicon octicon-link';
+        anchor.appendChild(iconDiv);
+      }
+
+      if (shouldScroll && scrollToHash(location.hash)) {
+        shouldScroll = false;
+      }
```

Comment:
Para conseguir redirecionar ao acessar um link com o `hash` que contém um `id` ao fim de uma página com muitos comentários, precisei confirmar se o `scrollToHash` havia encontrado o `id`. Isso funcionou localmente, mas não em homologação (não funcionou nem para uma âncora na publicação). Aceito sugestões para melhorar/corrigir isso. Acho que usar um `setTimeout` não é o ideal.

* Um exemplo no fim da página: https://tabnews-git-add-anchor-to-headers-tabnews.vercel.app/rafael/teste-de-ancoras-em-titulos#teste-259
* Na publicação: https://tabnews-git-add-anchor-to-headers-tabnews.vercel.app/rafael/teste-de-ancoras-em-titulos#segundo-nivel